### PR TITLE
[NEW][Apps] Remove TS compiler

### PR DESCRIPTION
**Pull request by d-gubert**
## Proposed changes

This removes the compilation process for the apps from Rocket.Chat. From this point on, we're expecting the apps package to contain their compiled versions already.

This is just part of the project - most of the work will be done on other packages.

Apps-Engine PR: https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/307

Code is reviewable, we're figuring out what Rocket.Chat version it should be released in, as it requires other operational changes in the Marketplace API (that's why this is still a draft)

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by thassiov** [comment]
LGTM

**Review by lolimay — APPROVED** [review]

**Comment by d-gubert** [comment]
We still need to change the URL we're going to use to download the apps from the Marketplace, as there will be a new one serving only the compiled version.

File: app/apps/server/communication/rest.js [hunk 1]
```diff
@@ -433,7 +429,7 @@ export class AppsRestApi {
 
 					let result;
 					try {
-						result = HTTP.get(`${ baseUrl }/v1/apps/${ this.bodyParams.appId }/download/${ this.bodyParams.version }`, {
+						result = HTTP.get(`${ baseUrl }/v2/apps/${ this.bodyParams.appId }/download/${ this.bodyParams.version }`, {
```
Comment (d-gubert) [inline app/apps/server/communication/rest.js:432]: @geekgonecrazy this should be enough, right?

Comment (geekgonecrazy) [reply app/apps/server/communication/rest.js:432]: 👍

**Review by geekgonecrazy — APPROVED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**